q.js

```diff
@@ -1645,6 +1645,9 @@ Q["finally"] = function (object, callback) {
 
 Promise.prototype.fin = // XXX legacy
 Promise.prototype["finally"] = function (callback) {
+    if (callback === undefined) {
```

Comment:
@willtcarey Good catch. Of course, `if (callback && typeof callback.apply !== "function")`.